robotiq_s_model_articulated_gazebo_plugins/src/RobotiqHandPlugin.cpp

```diff
@@ -0,0 +1,888 @@
+#undef private
+
+// Default topic names initialization.
+const std::string RobotiqHandPlugin::DefaultLeftTopicCommand  =
+  "/left_hand/command";
+const std::string RobotiqHandPlugin::DefaultLeftTopicState    =
+  "/left_hand/state";
+const std::string RobotiqHandPlugin::DefaultRightTopicCommand =
+  "/right_hand/command";
+const std::string RobotiqHandPlugin::DefaultRightTopicState   =
+  "/right_hand/state";
+
+////////////////////////////////////////////////////////////////////////////////
+RobotiqHandPlugin::RobotiqHandPlugin()
+{
+  // PID default parameters.
+  for (int i = 0; i < this->NumJoints; ++i)
+  {
+    this->posePID[i].Init(1.0, 0, 0.5, 0.0, 0.0, 60.0, -60.0);
+    this->posePID[i].SetCmd(0.0);
+  }
+
+  // Default grasping mode: Basic mode.
+  this->graspingMode = Basic;
+
+  // Default hand state: Disabled.
+  this->handState = Disabled;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+RobotiqHandPlugin::~RobotiqHandPlugin()
+{
+  gazebo::event::Events::DisconnectWorldUpdateBegin(this->updateConnection);
+  this->rosNode->shutdown();
+  this->rosQueue.clear();
+  this->rosQueue.disable();
+  this->callbackQueueThread.join();
+}
+
+////////////////////////////////////////////////////////////////////////////////
+void RobotiqHandPlugin::Load(gazebo::physics::ModelPtr _parent,
+                             sdf::ElementPtr _sdf)
+{
+  this->model = _parent;
+  this->world = this->model->GetWorld();
+  this->sdf = _sdf;
+
+  if (!this->sdf->HasElement("side") ||
+      !this->sdf->GetElement("side")->GetValue()->Get(this->side) ||
+      ((this->side != "left") && (this->side != "right")))
+  {
+    gzerr << "Failed to determine which hand we're controlling; "
+             "aborting plugin load. <Side> should be either 'left' or 'right'."
+          << std::endl;
+    return;
+  }
+
+  // Load the vector of all joints.
+  std::string prefix;
+  if (this->side == "left")
+    prefix = "l_";
+  else
+    prefix = "r_";
+
+  // Load the vector of all joints.
+  if (!this->FindJoints())
+    return;
+
+  gzlog << "Prior to iterating.." << std::endl;
+  // Initialize joint state vector.
+  this->jointStates.name.resize(this->jointNames.size());
+  this->jointStates.position.resize(this->jointNames.size());
+  this->jointStates.velocity.resize(this->jointNames.size());
+  this->jointStates.effort.resize(this->jointNames.size());
+  gzlog << "About to iterate things.." << std::endl;
+  for (size_t i = 0; i < this->jointNames.size(); ++i)
+  {
+    this->jointStates.name[i] = this->jointNames[i];
+    this->jointStates.position[i] = 0;
+    this->jointStates.velocity[i] = 0;
+    this->jointStates.effort[i] = 0;
+  }
+  gzlog << "Initialized the joint state vector" << std::endl;
+
+  // Default ROS topic names.
+  std::string controlTopicName = this->DefaultLeftTopicCommand;
+  std::string stateTopicName   = this->DefaultLeftTopicState;
+  if (this->side == "right")
```

Comment:
@TheDash There is also this notion of setting topic names based on "side".   Control topic name and state topic names should either be specified or match the default from running the controller with a single gripper if you really want a default.